### DISCUSSION TITLE
Cargo run not working

### DISCUSSION THREAD

**Discussion by AkshatShukla10** [juspay/hyperswitch#758] [Q&A] [answered]
After setting up all the dependencies on WSL Ubuntu and setting up database when I execute cargo run it is compiling almost all the libraries but then it cannot compile 'router' at the end.

**Comment by Narayanbhat166** [comment]
Can you please provide some more information as to what made `router` crate compile? attaching the logs and cargo version would be helpful. To get the cargo version, run `cargo --version`.

**Reply by AkshatShukla10** [reply]
![Screenshot (3)](https://user-images.githubusercontent.com/65565793/225634023-dc0ae1b0-dc59-457e-ab7d-abb5b231bbdf.png)

Cargo version is 1.65.0

**Comment by Narayanbhat166** [comment, marked as answer]
Seems like Out of memory issue as evident by the `SIGKILL` and this post https://github.com/rust-lang/rust/issues/100164. Can you check the memory limits for wsl and try increasing it and retry? To increase the memory limit in your .wslconfig add this configuration 
```config
[wsl2]
memory=2GB
```
https://fizzylogic.nl/2023/01/05/how-to-configure-memory-limits-in-wsl2
Also close any other open windows which are currently not being used if it fails again

**Reply by cb-sudheer** [reply]
I am using Mac OS, do we need to add any other configuration other than wsl2?

**Comment by ChinmayMoghe** [comment]
I am proposing a small change in local setup readme, addition of a note to try_local_system.md, that just mentions increasing memory limits in case users face issue in compilation while using WSL2 ?

Note: 

In case you are facing issues compiling on WSL2, try increasing the memory limits by configuring %USERPROFILE%/.wslconfig file

Example config:

[wsl2]
memory=12GB

**Comment by Thrilokgithub** [comment]
Seems like Out of memory issue as evident by the SIGKILL and this post https://github.com/rust-lang/rust/issues/100164. Can you check the memory limits for wsl and try increasing it and retry? To increase the memory limit in your .wslconfig add this configuration

[wsl2]
memory=2GB
https://fizzylogic.nl/2023/01/05/how-to-configure-memory-limits-in-wsl2
Also close any other open windows which are currently not being used if it fails again
